magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php

```diff
@@ -46,6 +51,7 @@ public function __construct($settings=[]) {
         $this->_region = isset($settings['s3_region'])? $settings['s3_region'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_region');
         $this->_bucket = isset($settings['s3_bucket']) ? $settings['s3_bucket'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_bucket');
         $this->_folder = isset($settings['s3_folder']) ? $settings['s3_folder'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_folder');
+        $this->_cachecontrol = isset($settings['s3_cachecontrol']) ? $settings['s3_cachecontrol'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_cachecontrol');
```

Comment:
it's coming out of magento config... maybe follow this line with        

if (!is_numeric($this->_cachecontrol)) unset($this->_cachecontrol);